igvm/vm.py

```diff
@@ -197,6 +201,46 @@ def start(self, force_stop_failed=True, transaction=None):
         if transaction:
             transaction.on_rollback('stop VM', self.shutdown)
 
+    def aws_start(self):
+        """AWS start
+
+        Start a VM in AWS.
+        """
+
+        ec2 = boto3.client('ec2')
+
+        try:
+            ec2.start_instances(
+                InstanceIds=[self.dataset_obj['aws_instance_id']],
+                DryRun=True
+            )
+        except ClientError as e:
+            if 'DryRunOperation' not in str(e):
+                raise
+
+        try:
+            response = ec2.start_instances(
+                InstanceIds=[self.dataset_obj['aws_instance_id']],
+                DryRun=False
+            )
+            current_state = response['StartingInstances'][0][
+                'CurrentState']['Code']
```

Comment:
Then the line is too long. What's your proposal?

Reply:
Hmm, you could:
- Just keep the line too long, sometimes that's the least bad option.
- Or do this:
```python
            current_state = (
                response['StartingInstances'][0]['CurrentState']['Code']
            )
```
- Or this:
```python
            response = ec2.start_instances(
                InstanceIds=[self.dataset_obj['aws_instance_id']],
                DryRun=False
            )['StartingInstances'][0]
            current_state = response['CurrentState']['Code']
```

Either way this makes a lot of assumptions about the data structure returned by boto. Such large datastructure accesses often highlight code places with too little error handling. It's easy to imagin error cases in which the api returns something weird and you run into an IndexError or KeyError here. I tried to have a look at the boto(core) code, but it's quite abstract so I couldn't come to a good conclusion whether this might be safe after all.

Reply:
fixed